src/main/java/org/prebid/server/cookie/UidsCookieService.java

```diff
@@ -250,43 +271,53 @@ private static UidsCookie removeExpiredUids(UidsCookie uidsCookie) {
         return updatedCookie;
     }
 
-    private UidsCookieUpdateResult updateUidsCookieByPriority(UidsCookie uidsCookie, String familyName, String uid) {
-        final UidsCookie updatedCookie = uidsCookie.updateUid(familyName, uid);
-        if (!cookieExceededMaxLength(updatedCookie)) {
-            return UidsCookieUpdateResult.updated(updatedCookie);
-        }
-
-        if (!prioritizedCoopSyncProvider.hasPrioritizedBidders()
-                || prioritizedCoopSyncProvider.isPrioritizedFamily(familyName)) {
-            return UidsCookieUpdateResult.updated(trimToLimit(updatedCookie));
-        } else {
-            metrics.updateUserSyncSizeBlockedMetric(familyName);
-            return UidsCookieUpdateResult.unaltered(uidsCookie);
-        }
-    }
-
-    private boolean cookieExceededMaxLength(UidsCookie uidsCookie) {
-        return maxCookieSizeBytes > 0 && cookieBytesLength(uidsCookie) > maxCookieSizeBytes;
-    }
+    public Map<String, UidsCookie> splitUids(UidsCookie uidsCookie) {
+        final Uids cookieUids = uidsCookie.getCookieUids();
+        final Map<String, UidWithExpiry> uids = cookieUids.getUids();
+        final boolean hasOptout = !uidsCookie.allowsSync();
+
+        final Iterator<String> cookieFamilyIterator = cookieFamilyNamesByDescPriorityAndExpiration(uidsCookie);
+        final Map<String, UidsCookie> splitCookies = new HashMap<>();
+
+        int uidsIndex = 0;
+        String nextCookieFamily = null;
+
+        while (uidsIndex < numberOfUidCookies) {
+            final String uidsName = uidsIndex == 0 ? COOKIE_NAME : COOKIE_NAME_FORMAT.formatted(uidsIndex + 1);
+            final UidsCookie tempUidsCookie = splitCookies.computeIfAbsent(
+                    uidsName,
+                    key -> new UidsCookie(Uids.builder().uids(new HashMap<>()).optout(hasOptout).build(), mapper));
+            final Map<String, UidWithExpiry> tempUids = tempUidsCookie.getCookieUids().getUids();
+
+            while (nextCookieFamily != null || cookieFamilyIterator.hasNext()) {
+                nextCookieFamily = nextCookieFamily == null ? cookieFamilyIterator.next() : nextCookieFamily;
+                tempUids.put(nextCookieFamily, uids.get(nextCookieFamily));
+                if (cookieExceededMaxLength(uidsName, tempUidsCookie)) {
+                    tempUids.remove(nextCookieFamily);
```

Comment:
Also, thats not how you update `UidsCookie`. Please, use `deleteUid` method. 
This model were intended to be immutable. I don't know why underlying hash map is not copied upon creation. I think it was an oversight.